File: 0026删除有序数组重复项.cpp
#include<vector>
#include<iostream>

using namespace std;

// 快慢指针
int RemoveDuplicates(vector<int>& nums)
{
    int slow = 0;
    int fast = 1;
    while(fast<nums.size())
    {
        if(nums[slow]!=nums[fast])
            nums[++slow] = nums[fast];
        fast++;
    }

    return slow + 1;
}

int main()
{
    vector<int> v = {1, 1, 2, 2, 3};
    int a=RemoveDuplicates(v);
    cout << a <<endl;

    for (int i = 0; i < v.size(); ++i)
    {
        cout << v[i];
    }
}

File: python/main/0026删除有序数组重复项.py
nums=[1,2,3,3,3]

# 快慢指针
def removeDuplicates(nums):
    if(not nums):
        return 0
    res_len=0
    
    for i in range(1,len(nums)):
        if(nums[res_len]!=nums[i]):
            res_len+=1
            nums[res_len]=nums[i]
    
    return res_len+1


print(removeDuplicates(nums))
print(nums)